<launch>

  <arg name="robot_model_master"                default="wx250s"/>
  <arg name="robot_model_puppet"                default="wx250s"/>

  <arg name="base_link_master"                  default="base_link"/>
  <arg name="base_link_puppet"                  default="base_link"/>

  <arg name="master_modes_left"                 default="$(find single_arm)/config/master_modes_left.yaml"/>
  <arg name="puppet_modes_left"                 default="$(find single_arm)/config/puppet_modes_left.yaml"/>

  <arg name="launch_driver"                     default="true"/>
  <arg name="use_sim"                           default="false"/>

  <arg name="robot_name_master_left"            value="master_left"/>
  <arg name="robot_name_puppet_left"            value="puppet_left"/>



  <include if="$(arg launch_driver)" file="$(find interbotix_xsarm_control)/launch/xsarm_control.launch">
    <arg name="robot_model"                       value="$(arg robot_model_master)"/>
    <arg name="robot_name"                        value="$(arg robot_name_master_left)"/>
    <arg name="base_link_frame"                   value="$(arg base_link_master)"/>
    <arg name="use_world_frame"                   value="false"/>
    <arg name="use_rviz"                          value="false"/>
    <arg name="mode_configs"                      value="$(arg master_modes_left)"/>
    <arg name="use_sim"                           value="$(arg use_sim)"/>
  </include>



  <include if="$(arg launch_driver)" file="$(find interbotix_xsarm_control)/launch/xsarm_control.launch">
    <arg name="robot_model"                       value="$(arg robot_model_puppet)"/>
    <arg name="robot_name"                        value="$(arg robot_name_puppet_left)"/>
    <arg name="base_link_frame"                   value="$(arg base_link_puppet)"/>
    <arg name="use_world_frame"                   value="false"/>
    <arg name="use_rviz"                          value="false"/>
    <arg name="mode_configs"                      value="$(arg puppet_modes_left)"/>
    <arg name="use_sim"                           value="$(arg use_sim)"/>
  </include>



  <node
    name="master_left_transform_broadcaster"
    pkg="tf2_ros"
    type="static_transform_publisher"
    args="0 -0.25 0 0 0 0 /world /$(arg robot_name_master_left)/base_link"/>

  <node
    name="puppet_left_transform_broadcaster"
    pkg="tf2_ros"
    type="static_transform_publisher"
    args="0 -0.25 0 0 0 0 /world /$(arg robot_name_puppet_left)/base_link"/>


</launch>